### ISSUE TITLE
[Issue] ExoticGarden Fruits Not Working

### ISSUE THREAD

**Issue by Zykana** [DieReicheErethons/Brewery#414]
So ExoticGarden fruits are unable to be put into cauldrons. No errors or anything, just won't do it.

I am using Spigot 1.17, and Exotic Garden Dev 76.

**Comment by Sn0wStorm** [comment]
Did you put the id of the fruits into a recipe?

**Comment by Zykana** [comment]
Yes, did how we format the ID change?

**Comment by Sn0wStorm** [comment]
No unless something in SlimeFun or Exotic Garden changed, it should still be the same. I'll look into it

**Comment by Zykana** [comment]
I'm unable to use even non-slimefun heads. I think perhaps a plugin might be interfering here.

**Comment by SFGoose** [comment]
I'm having the same issue, did you ever figure out the problem?

**Comment by Zykana** [comment]
> I'm having the same issue, did you ever figure out the problem?

No I did not. :(

**Comment by CloneCluke** [comment]
I am also having this problem; Normal Slimefun items appear to be working, but none of the exotic garden. Anyone have a fix? I use exotic garden in close to 200 of my recipes.

**Comment by ElNico56** [comment]
i'm having the same problem, no slimefun items work!

**Comment by TristanVdElzen** [comment]
I've done some testing and it appears that all non-vanilla items that are consumable can no longer be added to cauldrons.
This includes:
- brews from the brewery plugin
- Slimefun consumable items
- ExoticGardens consumable items (e.g. yeast and cornmeal works fine, but grapes or grape juice does not)

My thinking is that the right-click event starts item consumption without checking if the target block is interactable.

**Comment by CloneCluke** [comment]
> 
> 
> I've done some testing and it appears that all non-vanilla items that are consumable can no longer be added to cauldrons. This includes:
> 
>     * brews from the brewery plugin
> 
>     * Slimefun consumable items
> 
>     * ExoticGardens consumable items (e.g. yeast and cornmeal works fine, but grapes or grape juice does not)
> 
> 
> My thinking is that the right-click event starts item consumption without checking if the target block is interactable.
> 
> Edit: solving this issue would likely also solve #421 as it pertains to the same operation.

I noticed the same thing; however there was one particular consumable from Slimefun that worked for some reason. Still have not managed to find a work around

**Comment by ElNico56** [comment]
It's not interactivity, since brewery brews can be used in further recipes (tested using last update on both as of the writting) it's slimefun specifically that breaks

**Comment by TristanVdElzen** [comment]
> It's not interactivity, since brewery brews can be used in further recipes (tested using last update on both as of the writting) it's slimefun specifically that breaks

You are right! I tested some more, turns out my mistake with the brews was that I used the brew ID in my recipes. -facepalm- When an existing brew is used in a recipe specified by **name** it will work. :)

Issue remains for Slimefun and ExoticGarden consumable items, which as you pointed out is a fault in their code. Using those items on the cauldron does not trigger the Brewery `clickCauldron()` function.

**Comment by TristanVdElzen** [comment]
Fixed in ExoticGarden pull request: https://github.com/TheBusyBiscuit/ExoticGarden/pull/217
Requires Brewery pull request #435 in order to work.

**Comment by Sn0wStorm** [comment]
Im guessing this commit also fixes the issue. https://github.com/DieReicheErethons/Brewery/commit/db428892e91e9565e50ce4123ed2afbb70eac68e
